hack/cloud-image-uploader/azure.go

```diff
@@ -0,0 +1,420 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/pageblob"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"log"
+	"math"
+	"os"
+	"strings"
+	"sync/atomic"
+)
+
+
+var (
+	subscriptionID       string
+	cred                 *azidentity.DefaultAzureCredential
+	authorizer           autorest.Authorizer
+	storageClientFactory *armstorage.ClientFactory
+	accountsClient       *armstorage.AccountsClient
+	providersClient      resources.ProvidersClient
+	locations         = make(map[string]Location)
+	resourceGroupName = "SideroGallery"
+	defaultRegion = "eastus"
+	galleryName = "SideroLabs"
```

Comment:
the bottom part should be `const` probably, while the top part of variables should better be part of the `AzureUploader` struct and used as members, this provides better scoping

In general, we shouldn't use global variables if possible